hoot-core/src/main/cpp/hoot/core/io/schema/Schema.cpp

```diff
@@ -35,17 +35,36 @@ Schema::Schema()
 {
 }
 
+void Schema::addLayer(boost::shared_ptr<Layer> l)
+{
+  _layers.push_back(l);
+  _layerNameMap[l->getName()] = _layers.size()-1;
```

Comment:
This is great having both a layers and layer name map data store, way better look-up times.